source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -1474,15 +1479,14 @@ SwiftLanguageRuntime::GetMemberVariableOffset(CompilerType instance_type,
     // Bind generic parameters if necessary.
     if (instance && swift_type->hasTypeParameter())
       if (auto *frame = instance->GetExecutionContextRef().GetFrameSP().get())
-        if (auto bound = DoArchetypeBindingForType(*frame, instance_type)) {
+        if (auto bound = DoArchetypeBindingForType(
+                *frame, instance->GetStaticValue()->GetCompilerType())) {
```

Comment:
Can we consider adding a comment here explaining why we do the archetype binding on the static type?